File: praca-inzynierska-frontend/src/app/app/model/dto/WorkingHoursDto.ts
import { TimePeriodDto } from './TimePeriodDto'
import { WorkingHoursBackendDto } from './WorkingHoursBackendDto'

export class WorkingHoursDto {
    
    monday: TimePeriodDto
    tuesday: TimePeriodDto
    wednesday: TimePeriodDto
    thursday: TimePeriodDto
    friday: TimePeriodDto
    saturday: TimePeriodDto
    sunday: TimePeriodDto

    constructor(backendDto: WorkingHoursBackendDto) {
        this.monday = backendDto.monday ? TimePeriodDto.of(backendDto.monday) : null
        this.tuesday = backendDto.tuesday ? TimePeriodDto.of(backendDto.tuesday) : null
        this.wednesday = backendDto.wednesday ? TimePeriodDto.of(backendDto.wednesday) : null
        this.thursday = backendDto.thursday ? TimePeriodDto.of(backendDto.thursday) : null
        this.friday = backendDto.friday ? TimePeriodDto.of(backendDto.friday) : null
        this.saturday = backendDto.saturday ? TimePeriodDto.of(backendDto.saturday) : null
        this.sunday = backendDto.sunday ? TimePeriodDto.of(backendDto.sunday) : null
    }
}

File: praca-inzynierska-backend/src/main/kotlin/jwer/pracainzynierskabackend/model/dto/WorkingHoursDto.kt
package jwer.pracainzynierskabackend.model.dto

import jwer.pracainzynierskabackend.model.embeddable.TimePeriod
import jwer.pracainzynierskabackend.model.entity.WorkingHours

data class WorkingHoursDto(
        val monday: TimePeriod?,
        val tuesday: TimePeriod?,
        val wednesday: TimePeriod?,
        val thursday: TimePeriod?,
        val friday: TimePeriod?,
        val saturday: TimePeriod?,
        val sunday: TimePeriod?
) {
    constructor(wh: WorkingHours) :
            this(
                    wh.monday,
                    wh.tuesday,
                    wh.wednesday,
                    wh.thursday,
                    wh.friday,
                    wh.saturday,
                    wh.sunday
            )
}